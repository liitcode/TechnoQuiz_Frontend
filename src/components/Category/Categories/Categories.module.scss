@import '../../../assets/styles/mixins.scss';
.categories {
  width: 95%;
  max-height: 398px;
  overflow-y: scroll;

  &__cards {
    @include set-flex;
    flex-wrap: wrap;
    justify-content: flex-start;
  }
}

@media only screen and (min-width: 768px) {
  .categories {
    width: 70%;
    padding: 3%;
    max-height: unset;
    overflow-y: unset;
  }
}
