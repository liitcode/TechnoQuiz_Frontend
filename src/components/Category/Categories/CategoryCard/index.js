/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable no-alert */
import { motion, useMotionValue, useTransform } from 'framer-motion';
import React from 'react';
import { useDispatch } from 'react-redux';
import { openTypeSelectionModal } from '../../../../Redux/actions/actionCreators/category';

import styles from './CategoryCard.module.scss';

function CategoryCard() {
  const x = useMotionValue(0);
  const y = useMotionValue(0);
  const rotateX = useTransform(y, [-100, 100], [30, -30]);
  const rotateY = useTransform(x, [-100, 100], [-30, 30]);

  // dispatch(openTypeSelectionModal)

  const dispatch = useDispatch();

  return (
    <motion.div
      onClick={() => dispatch(openTypeSelectionModal('JAVASCRIPT'))}
      className={styles.card}
      style={{ x, y, rotateX, rotateY, z: 100 }}
      drag
      dragElastic={0.16}
      dragConstraints={{ top: 0, left: 0, right: 0, bottom: 0 }}
      whileTap={{ cursor: 'grabbing' }}
    >
      <div className={styles.card__inner}>
        <div className={styles.card__front}>
          <div className={styles.card__front__image} />
        </div>
        <div className={styles.card__back}>
          <h1>Click here to take quiz on PYTHON</h1>
        </div>
      </div>
    </motion.div>
  );
}

export default CategoryCard;
