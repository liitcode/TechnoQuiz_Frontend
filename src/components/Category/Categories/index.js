/* eslint-disable no-unused-vars */
import React, { useState, useEffect } from 'react';
import axios from 'axios';
import CategoryCard from './CategoryCard';
import styles from './Categories.module.scss';

function Categories() {
  const [data, setData] = useState({});
  useEffect(async () => {
    const result = await axios(
      'https://technoquiz-env.eba-33dpsiuk.ap-south-1.elasticbeanstalk.com/api/category',
    );
    setData(result.data);
  }, []);

  return (
    <div className={styles.categories}>
      <h2>Choose Category</h2>
      <div className={styles.categories__cards}>
        {Object.keys(data).map((item) => (
          <CategoryCard key={item} cat={data[item]} />
        ))}
      </div>
    </div>
  );
}

export default Categories;
