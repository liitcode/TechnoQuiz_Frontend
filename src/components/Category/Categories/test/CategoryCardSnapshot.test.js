import React from "react";
import renderer from "react-test-renderer";
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import storeConfig from '``../../../Redux/store/configureStore';
import Categories from "../index";

it('renders the component', () => {
    const tree = renderer.create(<Provider store={storeConfig}>
        <MemoryRouter><Categories /></MemoryRouter></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
})