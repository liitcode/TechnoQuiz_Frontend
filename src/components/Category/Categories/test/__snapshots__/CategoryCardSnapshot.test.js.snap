// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  className="categories"
>
  <h2>
    Choose Category
  </h2>
  <div
    className="categories__cards"
  >
    <div
      className="skeleton-card"
    >
      <div
        className="skeleton-circle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-rectangle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-card"
    >
      <div
        className="skeleton-circle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-rectangle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-card"
    >
      <div
        className="skeleton-circle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-rectangle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-card"
    >
      <div
        className="skeleton-circle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-rectangle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-card"
    >
      <div
        className="skeleton-circle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-rectangle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-card"
    >
      <div
        className="skeleton-circle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-rectangle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-card"
    >
      <div
        className="skeleton-circle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-rectangle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-card"
    >
      <div
        className="skeleton-circle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-rectangle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-card"
    >
      <div
        className="skeleton-circle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-rectangle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-card"
    >
      <div
        className="skeleton-circle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-rectangle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-card"
    >
      <div
        className="skeleton-circle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-rectangle"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
